src/js/drawer.js

```diff
@@ -0,0 +1,139 @@
+	let timeout;
+
+	const handleKeydown = (e) => {
+		if (e.keyCode === 27) {
+			callback();
+		}
+	};
+
+	const handleClick = (e) => {
+		if (!scope.contains(e.target)) {
+			console.log('callbacking');
+			callback();
+		}
+	};
+
+	const handleMouseenter = () => {
+		clearTimeout(timeout);
+	};
+
+	const handleMouseleave = () => {
+		timeout = setTimeout(callback, INTENT_DELAY);
+	};
+
+	const removeEvents = () => {
+		console.log('removing events');
+		clearTimeout(timeout);
+
+		scope.removeEventListener('mouseenter', handleMouseenter);
+		scope.removeEventListener('mouseleave', handleMouseleave);
+		document.removeEventListener('click', handleClick);
+		document.removeEventListener('touchstart', handleClick);
+		document.removeEventListener('keydown', handleKeydown);
+	};
+
+	const addEvents = () => {
+		console.log('adding events');
+		scope.addEventListener('mouseenter', handleMouseenter);
+		scope.addEventListener('mouseleave', handleMouseleave);
+		document.addEventListener('click', handleClick);
+		document.addEventListener('touchstart', handleClick);
+		document.addEventListener('keydown', handleKeydown);
+	};
+
+	return { addEvents, removeEvents };
+}
+
+function addDrawerToggles (headerEl, drawerEl) {
+	const controls = [];
+	controls.push(...headerEl.querySelectorAll(`[aria-controls="${drawerEl.id}"]`));
+	controls.push(...drawerEl.querySelectorAll(`[aria-controls="${drawerEl.id}"]`));
+
+	let handleClose;
+	function toggleCallback(state) {
+		if (state === 'close') {
+			handleClose.removeEvents();
+		} else {
+			// don't capture the initial click or accidental double taps etc.
+			// we could use transitionend but scoping is tricky and it needs prefixing and...
+			setTimeout(handleClose.addEvents, LISTEN_DELAY);
+		}
+
+		drawerEl.classList.toggle('o-header__drawer--closing', state === 'close');
+		drawerEl.classList.toggle('o-header__drawer--opening', state === 'open');
+	}
+
+	controls.forEach((control) => {
+		const drawerToggle = new Toggle(control, {
+			target: drawerEl,
+			callback: toggleCallback
+		});
+
+		// Both toggles have the same target, so the toggle function will be the same
+		// If there's a separate handleClose instance for each toggle, removeEvents doesn't work
+		// when the close toggle is clicked
+		if (!handleClose) {
+			handleClose = handleCloseEvents(drawerEl, drawerToggle.toggle);
+		}
+	});
+}
+
+function addSubmenuToggles (drawerEl) {
+	const submenus = drawerEl.querySelectorAll('[id^="o-header-drawer-child-"]');
+
+	Array.from(submenus).forEach(submenu => {
+		const button = drawerEl.querySelector(`[aria-controls="${submenu.id}"]`);
+
+		submenu.setAttribute('aria-hidden', 'true');
+
+		new Toggle(button, {
+			target: submenu,
+			callback: (state) => {
+				button.textContent = button.textContent.replace(/fewer|more/, state === 'open' ? 'fewer' : 'more');
+			}
+		});
+	});
+}
+
+function appendUserDetail (target) {
+	return fetch('https://session-next.ft.com/', {
```

Comment:
Ok, it will stay as is for now then until that changes

On Mon, 18 Jul 2016 at 10:38 Matt Hinchliffe <notifications@github.com>
wrote:

> In src/js/drawer.js
> <https://github.com/Financial-Times/o-header/pull/188#discussion_r71120823>
> :
>
> > +	Array.from(submenus).forEach(submenu => {
> > +		const button = drawerEl.querySelector(`[aria-controls="${submenu.id}"]`);
> > +
> > +		submenu.setAttribute('aria-hidden', 'true');
> > +
> > +		new Toggle(button, {
> > +			target: submenu,
> > +			callback: (state) => {
> > +				button.textContent = button.textContent.replace(/fewer|more/, state === 'open' ? 'fewer' : 'more');
> > +			}
> > +		});
> > +	});
> > +}
> > +
> > +function appendUserDetail (target) {
> > +	return fetch('https://session-next.ft.com/', {
>
> https://financialtimes.slack.com/archives/ft-next-dev/p1468834128000030
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Financial-Times/o-header/pull/188/files/282752ebdb4d0e66e51fce59bdf1b74e7c87dbc9#r71120823>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA3u3edLw6zsjsF9AYQ8O-i0Z0uG7hfYks5qW0lcgaJpZM4JLWUy>
> .
>

-- 

------------------------------

*This email was sent by a company owned by Financial Times Group Limited 
("FT Group <http://aboutus.ft.com/corporate-information/#axzz3rajCSIAt>"), 
registered office at Number One Southwark Bridge, London SE1 9HL.  
Registered in England and Wales with company number 879531. This e-mail may 
contain confidential information. If you are not the intended recipient, 
please notify the sender immediately, delete all copies and do not 
distribute it further.  It could also contain personal views which are not 
necessarily those of the FT Group.  We may monitor outgoing or 
incoming emails as permitted by law.*